### PR TITLE
Docs for Blitz RPC endpoint to the app directory

### PR DESCRIPTION
**Pull request by siddhsuresh**
Docs for  https://github.com/blitz-js/blitz/pull/4341

### CODE REVIEW

File: app/pages/docs/rpc-setup.mdx [hunk 1]
```diff
@@ -47,7 +47,55 @@ export { withBlitz }
...
+})
+
+export {
+  invoke,
+}
+```
+
+4. Add Blitz RPC API Route:
+
+4.1 App router (recommended):
```
Comment (flybayer) [inline app/pages/docs/rpc-setup.mdx:76]: formatting here doesn't look good. Let's add clear headings like on the auth page

File: app/pages/docs/rpc-setup.mdx [hunk 2]
```diff
@@ -47,7 +47,55 @@ export { withBlitz }
...
+
+Create an `app/api/rpc` directory in your `src` directory with
+`[[...blitz]]/route.ts` file, and add the following lines:
```
Comment (flybayer) [inline app/pages/docs/rpc-setup.mdx:79]:
```suggestion
Create an `app/api/rpc/[[...blitz]]` directory in your `src` directory with
a `route.ts` file, and add the following lines:
```
```suggestion
Create an `app/api/rpc` directory in your `src` directory with
`[[...blitz]]/route.ts` file, and add the following lines:
```
```suggestion
Create an `app/api/rpc` directory in your `src` directory with
`[[...blitz]]/route.ts` file, and add the following lines:
```

**Review by flybayer — COMMENTED** [review]
see comments

**Review by flybayer — CHANGES_REQUESTED** [review]
somewhere you need to fully document the api, arguments, and return  of both `withBlitzAuth` and `rpcAppHandler`

File: app/pages/docs/rpc-setup.mdx (hunk 1 above)
Comment (siddhsuresh) [reply app/pages/docs/rpc-setup.mdx]: updated

File: app/pages/docs/rpc-config.mdx [hunk 3]
```diff
@@ -99,32 +101,104 @@ logging?: {
...
+
 <Card type="note">
-  Blitz RPC defaults to:
-  - `verbose` to be true if it not configured.
-  - The `input` and `resolver completion time` will be logged with the `info` level.
-  - The `result` and `next.js serialization time` will be logged with the `debug` level.
+  Blitz RPC defaults to: - `verbose` to be true if it not configured. -
+  The `input` and `resolver completion time` will be logged with the
+  `info` level. - The `result` and `next.js serialization time` will be
+  logged with the `debug` level.
```
Comment (flybayer) [inline app/pages/docs/rpc-config.mdx:109]: list formatting is broken

File: app/pages/docs/auth-server.mdx [hunk 4]
```diff
@@ -54,7 +54,104 @@ function PageWithGssp(props: Props) {
...
+
+```ts
+function withBlitzAuth(handlers: { [method: string]: Handler })
+```
+
+##### Signature
+
+##### Arguments
+
+- `handlers: { [method: string]: Handler })` - An object of handlers where
```
Comment (flybayer) [inline app/pages/docs/auth-server.mdx:91]: all the list formats seem broken

File: app/pages/docs/auth-server.mdx [hunk 5]
```diff
@@ -54,7 +54,104 @@ function PageWithGssp(props: Props) {
...
+#### `withBlitzAuth` API {#with-blitz-auth-api}
+
+The function supports both single handler as an input as well as an object
+of handlers and has the following signature:
+
+```ts
+function withBlitzAuth(handlers: { [method: string]: Handler })
+```
+
+##### Signature
```
Comment (flybayer) [inline app/pages/docs/auth-server.mdx:87]: this seems out of place?
```suggestion
```

**Review by flybayer — COMMENTED** [review]
otherwise the content looks good

**Pull request merged**